lib/IAMClient.js

```diff
@@ -1014,7 +1021,7 @@ class VaultClient {
      * @param {string} [contentType] - content type of body
      * @returns {undefined}
      */
-    request(method, path, iamAuthenticate, callback, data, reqUid,
+    async request(method, path, iamAuthenticate, callback, data, reqUid,
```

Comment:
Adding the `async` tag changes the return to a Promise, which in the case of `signer.sign` throwing would result in a `UnhandlePromiseRejection` and crash the process.

I'd suggest adding a `_signRequest` helper that is async and then using `then` and `catch` in request.
Something like:

```
async _signRequest(iamAuthenticate, req, options) 
```

That could then be used in request like
```
this._signRequest(iamAuthenticate, req, options)
    .catch(callback)
    .then(() => {
          ...
    });
```

Reply:
The helper was created, and the logic structure was reworked a bit to keep a flat function:
https://github.com/scality/vaultclient/pull/431/commits/1ce56261a91fa870df09cf0af9be14f1b7f12356